internal/state/execution.go

```diff
@@ -310,29 +310,33 @@ func (blockExec *BlockExecutor) ApplyBlock(
 	return state, nil
 }
 
-func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) (types.VoteExtension, error) {
+func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) ([]byte, error) {
 	req := abci.RequestExtendVote{
-		Vote: vote.ToProto(),
+		Hash:   vote.BlockID.Hash,
+		Height: vote.Height,
 	}
 
 	resp, err := blockExec.appClient.ExtendVote(ctx, req)
 	if err != nil {
-		return types.VoteExtension{}, err
+		return nil, err
```

Comment:
That's a good question, I think it would be fine to follow up in future PRs. It looks like the behavior is pretty inconsistent in this code, so it may take a bit of reworking.
I've opened https://github.com/tendermint/tendermint/issues/8206
to track.